document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/data/accounts/AccountsManager.java

```diff
@@ -299,6 +300,45 @@ public SmallFullAccountIxbrl getSmallFullAccounts(String link, Transaction trans
                 smallFullApiData.setOffBalanceSheet(offBalanceSheetApi);
             }
 
+            if (!StringUtils.isEmpty(smallFull.getLinks().getLoansToDirectors())) {
+
+                errorString = "loans to directors";
+
+                LoansToDirectorsApi loansToDirectors =
+                        apiClient.smallFull().loansToDirectors()
+                                .get(smallFull.getLinks().getLoansToDirectors()).execute().getData();
+
+                smallFullApiData.setLoansToDirectors(loansToDirectors);
+
+                if (!loansToDirectors.getLoans().isEmpty()) {
+
+                    errorString = "loans";
+
+                    // Find the first loan id from the map of loans
+                    String loanId =
+                            loansToDirectors.getLoans()
+                                    .keySet().stream().findFirst()
+                                    .orElseThrow(RuntimeException::new);
+
+                    // Use the loan id to derive a loan self link from the map of loans
+                    String loanSelfLink = loansToDirectors.getLoans().get(loanId);
+
+                    // Trim the end of the self link to calculate a generic 'loans' link, used to fetch all loans
+                    String loansLink = loanSelfLink.substring(0, loanSelfLink.lastIndexOf('/'));
```

Comment:
If we are having to monkey around with URL structures, is there something missing from our API?

Reply:
Good point, worth considering. Will revisit if we have time 👍